includes/functions.php

```diff
@@ -118,12 +119,14 @@ function generate_toc($html_string){
       $name = trim(str_replace('&nbsp;','', htmlentities(strip_tags($h_content) )));
       if($level > $curr_level){
         $toc .= "\n".'<div class="list-group">'."\n";
+        $counter +=1;
       } else if($level == $curr_level) {
         $toc .= "\n";
       } else {
         while($level < $curr_level){
           $toc .= "\n</div>\n\n";
           $curr_level -= 1;
+          $counter -=1;
         }
       }
       $curr_level = $level;
```

Comment:
Presumably wouldn't need this if we are changing `$curr_level` in the new `while` loop in the first `if` block..

```suggestion
```